### DISCUSSION TITLE
package extras change?

### DISCUSSION THREAD

**Discussion by josumoreno-BP** [sqlalchemy/sqlalchemy#10823] [Usage Questions] [answered]
Hello,

I was installing package extras until version 2.0.23 like _pip install "SQLAlchemy[postgresql_asyncpg]==2.0.23"_ i.e., _postres_asyncpg_ uses underscore. Since version 2.0.24 it needs to be hyphen instead of underscore, i.e. _postres-asyncpg_. If I go to pypi and download the sources for both versions I can see that PKG-INFO has changed those values although the setup.cfg is the same on both. Can you help me understand the problem?

Thanks

PS: I've compared both tags on github and I don't see any direct change for this, either any comment on releases about it

**Comment by zzzeek** [comment]
the tooling has changed, I think https://peps.python.org/pep-0685/ might have some high level detail, maybe someone else can fill in when exactly things changed

**Reply by CaselIT** [reply, marked as answer]
my understanding was that the old underscored version should still be installable by pip. I guess not...

If that's not the case (meaning they decided to break every underscore name) then there is no point in merging this https://github.com/sqlalchemy/sqlalchemy/pull/10481

**Comment by zzzeek** [comment]
i cant reproduce this, what is the exact version of pip in use and the commands you are running where you are seeing whatever messaging indicating "it needs to be hyphen instead of underscore" 

here's the latest pip 23.3.2:

```
$ pip install sqlalchemy[postgresql_asyncpg]
Requirement already satisfied: sqlalchemy[postgresql_asyncpg] in ./lib (2.0.25.dev0)
Requirement already satisfied: greenlet!=0.4.17 in /home/classic/.venv3/lib64/python3.12/site-packages (from sqlalchemy[postgresql_asyncpg]) (3.0.3)
Requirement already satisfied: typing-extensions>=4.6.0 in /home/classic/.venv3/lib64/python3.12/site-packages (from sqlalchemy[postgresql_asyncpg]) (4.8.0)
Collecting asyncpg (from sqlalchemy[postgresql_asyncpg])
  Downloading asyncpg-0.29.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.4 kB)
Downloading asyncpg-0.29.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.3 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.3/3.3 MB 1.7 MB/s eta 0:00:00
Installing collected packages: asyncpg
Successfully installed asyncpg-0.29.0

```

**Reply by josumoreno-BP** [reply]
Hey,

So first of all, I got this problem with poetry 1.1.15. Then I switched to raw pip to try to reproduce the problem and I thought I did it, but today I tried again and seems that I cannot. Checking the commands that I run I see that I made the mistake with _pip install "SQLAlchemy[postgresql-asyncpg]==2.0.24"_, and I did not read the message `WARNING: sqlalchemy 2.0.24 does not provide the extra 'postgresql-asyncpg'`, so that is why it did not install the extras. I guess this is correct and we should still use the underscore one (except on our poetry at least).

**Reply by CaselIT** [reply]
ok then the fallbacks that #10481 should be added also on v2 since it seems that some tools already do change the values specified in the extra
